### PR TITLE
Add vote revealer

### PR DESCRIPTION
**Pull request by mkot2**
didn't test this so tell me if there are any problems

it outputs to console right now and it's just fine, but is there a way to output messages to chat and only have local player see them?

### CODE REVIEW

**Comment by Tothale** [comment]
I have tried in gun race, working good.

**Comment by cristyanul** [comment]
It works!

**Comment by KernCS** [comment]
> but is there a way to output messages to chat and only have local player see them?

https://www.unknowncheats.me/forum/counterstrike-global-offensive/171375-printing-text-chat.html

File: Osiris/Hacks/Misc.cpp [hunk 1]
```diff
@@ -950,6 +950,28 @@ void Misc::preserveKillfeed(bool roundStart) noexcept
...
 
+void Misc::voteRevealer(GameEvent* event = nullptr) {
+    if (!config->misc.revealVotes)
+        return;
+
+    if (event) {
+        switch (fnv::hashRuntime(event->getName())) {
+        case fnv::hash("vote_cast"):
+            //dont ask me why this is here
+            std::ostringstream str;
```
Comment (danielkrupinski) [inline Osiris/Hacks/Misc.cpp:961]: No need to use `std::ostringstream`.

File: Osiris/Hacks/Misc.cpp [hunk 2]
```diff
@@ -950,6 +950,28 @@ void Misc::preserveKillfeed(bool roundStart) noexcept
...
+            str << interfaces->entityList->getEntity(event->getInt("entityid"))->getPlayerName() << " : ";
```
Comment (danielkrupinski) [inline Osiris/Hacks/Misc.cpp:962]: Entity nullptr check is missing.

File: Osiris/Hacks/Misc.cpp [hunk 3]
```diff
@@ -950,6 +950,28 @@ void Misc::preserveKillfeed(bool roundStart) noexcept
     }
 }
 
+void Misc::voteRevealer(GameEvent* event = nullptr) {
```
Comment (danielkrupinski) [inline Osiris/Hacks/Misc.cpp:953]: `event` could be `GameEvent&` as the function does nothing when `event` is nullptr.

**Review by danielkrupinski — CHANGES_REQUESTED** [review]

**Review by danielkrupinski — APPROVED** [review]

**Comment by danielkrupinski** [comment]
Thank you for the contribution!

**Pull request merged**